### PR TITLE
Enable mocking of const

### PR DESCRIPTION
**Pull request by JvJefke**
This PR enables the mocking of const variables.
It uses Babel to parse the code and transform const to let variables.

It can solve the following issues: #95, #79

### CODE REVIEW

**Comment by JvJefke** [comment]
The tests are failing because of the fact that older node versions don't support ES6 syntax. Is there a way to only run these tests in node 6.x versions?

**Comment by JvJefke** [comment]
@jhnns Could you give some input here? thx!

File: package.json [hunk 1]
```diff
@@ -36,7 +36,11 @@
   },
   "license": "MIT",
   "scripts": {
-    "test": "mocha -R spec --check-leaks",
+    "test": "mocha -R spec --check-leaks --globals __core-js_shared__",
     "coverage": "istanbul cover ./node_modules/mocha/bin/_mocha"
+  },
+  "dependencies": {
+    "babel-core": "~6.26.0",
```
Comment (jhnns) [inline package.json]: Any reason why you're not using the `^` operator? Personally, I prefer that.

Comment (JvJefke) [reply package.json]: I personally prefer to use fixed or `~`. A package should not contain breaking changes on minor updates but it often does. I find it more secure to make it fixed or set it to patch updates. Let me know if I still need to change is :).

**Comment by JvJefke** [comment]
I added the option because there may be use cases where the behaviour of const needs to be part of the test. This is probably an edge case though. Maybe we can flip the default behaviour and make it replace const by default?

**Comment by JvJefke** [comment]
You're right about the option. Hadn't thought about that. I'll remove the option when I have time.

**Comment by JvJefke** [comment]
I removed the options. It will now always use babel to convert const to let. I had to keep the old functionality though for `.coffee` scripts. Babel isn't able to convert it properly. I'll try to add the functionality for coffee scripts in the future.

File: testLib/sharedTestCases.js [hunk 2]
```diff
@@ -372,4 +372,51 @@ describe("rewire " + (typeof testEnv === "undefined"? "(node)": "(" + testEnv +
         revert();
     });
 
+    it("Should be possible to mock a const variable using __with__ syntax", function() {
+        var ES2015Module = rewire("./ES2015Module", {
+            convertConst: true
+        });
```
Comment (jhnns) [inline testLib/sharedTestCases.js]: Looks like you've forgotten this, but I'll clean it up, thanks :)

**Comment by jhnns** [comment]
Awesome 👍 
Published as 3.0.0

**Comment by vekexasia** [comment]
Hello, Any reason why this shouldn't work with a code written in typescript?
